loadtest/loadtest_client.go

```diff
@@ -108,194 +91,58 @@ func (c *LoadTestClient) Close() {
 	}
 }
 
-func (c *LoadTestClient) AppendTxHash(txHash string) {
-	c.TxResponseChan <- &txHash
-}
-
-func (c *LoadTestClient) WriteTxHashToFile() {
-	fmt.Printf("Writing Tx Hashes to: %s \n", c.TxHashFile.Name())
-	file := c.TxHashFile
-	for _, txHash := range c.TxHashList {
-		txHashLine := fmt.Sprintf("%s\n", txHash)
-		if _, err := file.WriteString(txHashLine); err != nil {
-			panic(err)
-		}
-	}
-}
-
-func (c *LoadTestClient) BuildTxs() (workgroups []*sync.WaitGroup, sendersList [][]func()) {
+func (c *LoadTestClient) BuildTxs(txQueue chan<- []byte, producerId int, numTxsPerProducerPerSecond int, wg *sync.WaitGroup, done <-chan struct{}, producedCount *int64) {
+	defer wg.Done()
+	ticker := time.NewTicker(1 * time.Second)
 	config := c.LoadTestConfig
-	numberOfAccounts := config.TxsPerBlock / config.MsgsPerTx * 2 // * 2 because we need two sets of accounts
-	activeAccounts := []int{}
-	inactiveAccounts := []int{}
-
-	for i := 0; i < int(numberOfAccounts); i++ {
-		if i%2 == 0 {
-			activeAccounts = append(activeAccounts, i)
-		} else {
-			inactiveAccounts = append(inactiveAccounts, i)
-		}
-	}
-
-	valKeys := c.SignerClient.GetValKeys()
-
-	for i := 0; i < int(config.Rounds); i++ {
-		fmt.Printf("Preparing %d-th round\n", i)
-
-		wg := &sync.WaitGroup{}
-		var senders []func()
-		workgroups = append(workgroups, wg)
-		c.generatedAdminMessageForBlock = false
-		for j, account := range activeAccounts {
-			accountIdentifier := fmt.Sprint(account)
-			accountKeyPath := c.SignerClient.GetTestAccountKeyPath(uint64(account))
-			key := c.SignerClient.GetKey(accountIdentifier, "test", accountKeyPath)
-
-			msgs, failureExpected, signer, gas, fee := c.generateMessage(config, key, config.MsgsPerTx)
-			txBuilder := TestConfig.TxConfig.NewTxBuilder()
-			_ = txBuilder.SetMsgs(msgs...)
-			seqDelta := uint64(i / 2)
-			mode := typestx.BroadcastMode_BROADCAST_MODE_SYNC
-			if j == len(activeAccounts)-1 {
-				mode = typestx.BroadcastMode_BROADCAST_MODE_BLOCK
+	accountIdentifier := fmt.Sprint(producerId)
+	accountKeyPath := c.SignerClient.GetTestAccountKeyPath(uint64(producerId))
+	key := c.SignerClient.GetKey(accountIdentifier, "test", accountKeyPath)
+	count := 0
+	for {
+		select {
+		case <-done:
+			fmt.Printf("Stopping producer %d\n", producerId)
+			return
+		case <-ticker.C:
+			count = 0
+		default:
+			if count < numTxsPerProducerPerSecond {
+				msgs, _, _, gas, fee := c.generateMessage(config, key, config.MsgsPerTx)
+				txBuilder := TestConfig.TxConfig.NewTxBuilder()
+				_ = txBuilder.SetMsgs(msgs...)
+				txBuilder.SetGasLimit(gas)
+				txBuilder.SetFeeAmount([]types.Coin{
+					types.NewCoin("usei", types.NewInt(fee)),
+				})
+				// Use random seqno to get around txs that might already be seen in mempool
+
+				c.SignerClient.SignTx(c.ChainID, &txBuilder, key, uint64(rand.Intn(math.MaxInt)))
```

Comment:
is this necessary after disabling seqno check? couldn't we just set them all to a constant seqno?

Reply:
I think you probably have to at least make these transactions different because mempool will reject duplicate txs.  However, even random seq can still have a chance to result in duplicates, we should probably just switch to a monitonically increasing seqno here?

Reply:
I wonder whether we need to apply rate limit for signing tx or not at all. We should apply the rate limiting only for the broadcast tx right? For signing txs, we can keep signing and generating tx as long as the queue/channel is not full